packages/flutter_tools/lib/src/macos/xcdevice.dart

```diff
@@ -564,6 +565,10 @@ class XCDevice {
           if (code != -10) {
             isConnected = false;
           }
+          // Error: iPhone is not paired with your computer. To use iPhone with Xcode, unlock it and choose to trust this computer when prompted. (code -9)
+          if (code == -9) {
```

Comment:
Re: the `ios-trust-failure` usage event above: I can't remember if there's a case where the error message is "not paired" but the error code is -9 (I imagine there was or we would have checked this code? Although the error parsing predates the code checking, I believe). So I left the behavior as-is.